vmdk_plugin/refcnt.go

```diff
@@ -203,6 +214,16 @@ func (r refCountsMap) decr(vol string) (uint, error) {
 	return rc.count, nil
 }
 
+// Return the associated mutex for the given volume
+func (r refCountsMap) getVolMtx(vol string) *sync.Mutex {
+	rc := r[vol]
+	if rc == nil {
+		rc = newRefCount()
+		r[vol] = rc
+	}
+	return rc.Mtx
+}
+
 // enumberates volumes and  builds refCountsMap, then sync with mount info
```

Comment:
Left them out for this, and mainly because the refCountsMap doesn't hold the responsibility of synchronisation. I have found another issues with this approach, comment below.